erpnext/hr/report/monthly_attendance_sheet/monthly_attendance_sheet.py

```diff
@@ -120,137 +138,258 @@ def get_chart_data(emp_att_map, days):
+				absent += 0.5
+		else:
+			absent +=  1
+	else:
+		unmarked += 1
+
+	return [present, leave, absent, holiday, unmarked]
+
+def get_abbr(status, status_map, leave_abbr_map, attendance_detail):
+	abbr = ''
+	abbr = status_map[status]['abbr']
+	if attendance_detail:
+		leave_type = attendance_detail[1]
+
+		if status_map[status]['is_leave'] and leave_type:
+			abbr = leave_abbr_map[leave_type]["full_day_abbr"]
+
+		if status_map[status]['is_half_day'] and leave_type:
+			abbr = leave_abbr_map[leave_type]["half_day_abbr"]
+			remaining_half_day_status = attendance_detail[2]
+
+			rem_half_day_abbr = status_map[remaining_half_day_status]['abbr']
+			abbr += " + " + rem_half_day_abbr
+	return abbr
+
+def get_status(attendance_detail, holiday_map, emp_holiday_list, day):
+	status = None
+	if attendance_detail:
+		status = attendance_detail[0]
+
+	if status is None and holiday_map:
+		if emp_holiday_list in holiday_map:
+			for idx, ele in enumerate(holiday_map[emp_holiday_list]):
+				if day == holiday_map[emp_holiday_list][idx][0]:
+					if holiday_map[emp_holiday_list][idx][1]:
+						status = "Weekly Off"
+					else:
+						status = "Holiday"
+
+	return status
+
+def get_attendance_status_abbr_map(get_abbr_map=0, remove_hard_coded_status = 0):
+	statuses = frappe.get_all("Attendance Status", fields= ['name', 'abbr', 'is_half_day', 'is_leave', 'is_present'])
+	statuses_map = {}
+
+	if get_abbr_map:
+		abbr_map = {}
+		for status in statuses:
+			abbr_map[status.abbr] = status
+		return abbr_map
+
+	for status in statuses:
+		statuses_map[status.name] = status
+
+	if not remove_hard_coded_status:
+		statuses_map["Weekly Off"] = {'abbr': "<b>WO</b>"}
+		statuses_map["Holiday"] = {'abbr': "<b>H</b>"}
+	return statuses_map
+
+def get_leave_type_abbr(get_abbr=0):
+	leave_type_abbrs = frappe.get_all("Leave Type", fields= ['name', 'full_day_abbr', 'half_day_abbr'])
+	leave_type_abbr_map = {}
+
+	if get_abbr:
+		full_day_abbr_list = [abbr.full_day_abbr for abbr in leave_type_abbrs]
+		half_day_abbr_list = [abbr.half_day_abbr for abbr in leave_type_abbrs]
+		return half_day_abbr_list, full_day_abbr_list
+
+	for abbr in leave_type_abbrs:
+		leave_type_abbr_map[abbr.name] = abbr
+
+	return leave_type_abbr_map
+
 def get_columns(filters):
 
 	columns = []
-
 	if filters.group_by:
 		columns = [_(filters.group_by)+ ":Link/Branch:120"]
-
 	columns += [
 		_("Employee") + ":Link/Employee:120", _("Employee Name") + ":Data/:120"
 	]
-	days = []
-	for day in range(filters["total_days_in_month"]):
-		date = str(filters.year) + "-" + str(filters.month)+ "-" + str(day+1)
-		day_name = day_abbr[getdate(date).weekday()]
-		days.append(cstr(day+1)+ " " +day_name +"::65")
+
+	to_date = getdate(filters["to_date"])
+	from_date = getdate(filters["from_date"])
+
+	if to_date < from_date:
+		frappe.throw(_("To date can not less than From date"))
+	if frappe.utils.date_diff(to_date, from_date) > 62:
+		frappe.throw(_("Please select interval period less than 2 months"))
```

Comment:
```suggestion
		frappe.throw(_("Please select the interval period between From Date and To Date to be less than 2 months"))
```

Any reason to limit the period?